src/controllers/users.controller.ts

```diff
@@ -35,11 +33,21 @@ export class UserController {
 
   @Post()
   async post(@Body() user: User): Promise<InsertResult> {
-    return this.usersService.createUser(user);
+    try {
+      const createdUser = await this.usersService.createUser(user);
+      return createdUser;
```

Comment:
```suggestion
      return await this.usersService.createUser(user);
```

Reply:
Not sure though, feel free to ignore it

Reply:
Actually `return await` is redundant. It should be just `return` in this case https://eslint.org/docs/rules/no-return-await

Reply:
Haha, so maybe we can change it in favor of just `return this.usersService.createUser(user);`

Reply:
Without `await`, the controller ignores the `try-catch` and always returns `500 Internal Server Error` instead of `400 Bad Request` with the corresponding message.

Reply:
Gottcha, it seems it's returning the whole promise without validating it. Makes sense, thanks 👍